checksum/checksum.go

```diff
@@ -0,0 +1,180 @@
+package checksum
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/schollz/progressbar/v3"
+)
+
+func checksumFor(path string) ([]byte, error) {
+	h := md5.New()
+
+	var r *os.File
+	r, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file in %s: %w", path, err)
+	}
+	defer r.Close()
+
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("error reading %s checksum content: %w", path, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("error reading %s content: %w", path, err)
```

(não estamos lendo o conteúdo do _checksum_, mas o conteúdo do arquivo)